lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -128,23 +139,17 @@ namespace Js
         }
         else
         {
-            int foundIndex = -1;
-            bool parentFound = this->otherParents != nullptr && this->otherParents->MapUntil([&](int index, RecyclerWeakReference<ArrayBufferParent>* item)
+            int foundCount = 0;
+            this->otherParents->MapAndRemoveIf([&](RecyclerWeakReference<ArrayBufferParent>* item)->bool
             {
-                if (item->Get() == parent)
+                if(item->Get() == nullptr)
                 {
-                    foundIndex = index;
+                    foundCount++;
                     return true;
                 }
                 return false;
-
             });
-
-            if (parentFound)
-            {
-                this->otherParents->RemoveAt(foundIndex);
-            }
-            else
+            if (foundCount != 1)
```

Comment:
sure. the compiler should optimize this away though..

---
In reply to: [99728161](https://github.com/Microsoft/ChakraCore/pull/2465#discussion_r99728161) [](ancestors = 99728161,99727464)